src/DotNetWorker.Grpc/Definition/GrpcFunctionDefinition.cs

```diff
@@ -71,16 +76,116 @@ public GrpcFunctionDefinition(FunctionLoadRequest loadRequest, IMethodInfoLocato
         {
             // Get the input converter attribute information, if present on the parameter.
             var inputConverterAttribute = parameterInfo?.GetCustomAttribute<InputConverterAttribute>();
-
             if (inputConverterAttribute != null)
             {
                 return new Dictionary<string, object>()
                 {
                     { PropertyBagKeys.ConverterType, inputConverterAttribute.ConverterType.AssemblyQualifiedName! }
                 }.ToImmutableDictionary();
             }
+            else
+            {
+                var inputAttribute = parameterInfo?.GetCustomAttribute<InputBindingAttribute>();
+                var triggerAttribute = parameterInfo?.GetCustomAttribute<TriggerBindingAttribute>();
+
+                return GetBindingAttributePropertiesDictionary(inputAttribute) ??
+                        GetBindingAttributePropertiesDictionary(triggerAttribute) ??
+                        ImmutableDictionary<string, object>.Empty;
+            }
+        }
+
+        private ImmutableDictionary<string, object>? GetBindingAttributePropertiesDictionary(BindingAttribute? bindingAttribute)
+        {
+            if (bindingAttribute is null)
+            {
+                return null;
+            }
+
+            var output = new Dictionary<string, object>();
+            bool isInputConverterAttributeAdvertised = false;
+
+            // ConverterTypesDictionary will be "object" part of the return value of this method - ImmutableDictionary<string, object>
+            // The dictionary has key of type IInputConverter and value as Properties of that converter (specifies supported types and support for Json Deserialization)
+            var converterTypesDictionary = new Dictionary<Type, ConverterProperties>();
+
+            IEnumerable<Attribute> customAttributes = bindingAttribute.GetType().GetCustomAttributes();
```

Comment:
instead of enumerating this twice, can we refactor to something like:

```suggestion
            Type type = bindingAttribute.GetType();
            if (type.GetCustomerAttribute<InputConverterAttribute>() is { } attribute)
            {
                // do something
                isInputConverterAttributeAdvertised = true;
            }
            
            if (isInputConverterAttributeAdvertised)
            {
                bool fallback = type.GetCustomAttribute<EnableConvertersFallbackAttribute>()?.Enabled ?? true;
            }
```

Reply:
Done